ssl/ssl_rsa.c

```diff
@@ -832,7 +855,7 @@ static int serverinfo_srv_add_cb(SSL *s, unsigned int ext_type,
             return 0;           /* No extension found, don't send extension */
         return 1;               /* Send extension */
     }
-    return -1;                  /* No serverinfo data found, don't send
+    return 0;                   /* No serverinfo data found, don't send
```

Comment:
As seen a few lines earlier, if no extension is found, don't send any extension _using a return value of 0_.  Thus using a return value of -1 here, for a similar case, is what leads to the failed handshake unexpectedly.